### PR TITLE
Catch all exceptions when installing app

### LINKED ISSUE #334: aiidalab does not detect interrupted app installation
If I trigger the installation of an app and interrupt it during the pip installation of dependencies,
aiidalab still subsequently says that the app is installed. I guess we need to have some kind of lockfile to make sure that installation finishes successfully?

Steps to reproduce: 

1. Launch installation with `aiidalab install`
2. Interrupt the installation in the middle
3. `aiidalab list` will show application as installed

### PR DESCRIPTION
**Pull request by danielhollas**
### Summary

Previously, we were only catching RuntimeError which left a lot of room for other failures during installation, which would result in an incomplete installation of an app. Generally, there's no way of knowing after-the-fact that the installation ended prematurely, e.g. `aiidalab list` would show the app as installed.

Note that this includes for example `post_install` script failures.

Now we catch everything, including Keyboard interrupts, so that we always trigger the already existing rollback mechanism.

See issue #334 for details of how to reproduce this.

### Test plan

1. Trigger installation of an app from CLI `aiidalab -v install aurora==0.2.1`
2. Interrupt the installation in the middle with Ctrl+C
3. The installation should rollback. Verify with `aiidalab list` that the app is not installed.
4. Try installing an app from a following URL, where I specifically modified the `post_install` script to fail. `aiidalab -v install aiidalab-ispg@git+https://github.com/ispg-group/aiidalab-ispg@post-install-fail`
5. Failing post_install should now trigger rollback as well.

Closes #334

Note that the more robust fix to this should be done in #364

### CODE REVIEW

**Review by unkcpz — COMMENTED** [review]
@danielhollas thanks for the fix, the changes look good. 

Since we have more experience with adding tests for this class, it is worth trying to monkeypatch a bunch of stuff in the `install` method and hint the exception to check it "work" as expected. If it is too complex we can leave it to the future.

**Comment by danielhollas** [comment]
Thanks for taking a look. I've looked into wrinting a test for the install workflow and it was a bit more work than I have time at the moment. I do not think it makes sense to test this functionality in isolation, we need to test the whole installation workflow comprehensively, and I do not have a bandwidth for that in the next three weeks unfortunately.

**Comment by danielhollas** [comment]
@yakutovicha can you take a brief look? Just want to verify that you agree with the approach I took here.

**Review by unkcpz — APPROVED** [review]
>  I do not think it makes sense to test this functionality in isolation, we need to test the whole installation workflow comprehensively, and I do not have a bandwidth for that in the next three weeks unfortunately.

No problem. I approved it, see if @yakutovicha want to have another look.

**Review by yakutovicha — APPROVED** [review]
I agree with the change 👍 . Please go ahead with merging it.

**Pull request merged**
